src/dlr_tvm.cc

```diff
@@ -164,6 +173,31 @@ void TVMModel::SetInput(const char* name, const int64_t* shape,
   UpdateInputShapes();
 }
 
+void TVMModel::SetInput(const char* name, DLTensor* tensor) {
+  std::string str(name);
+  int index = tvm_graph_runtime_->GetInputIndex(str);
+  if(index > -1) {
+    tvm::runtime::NDArray arr = tvm_graph_runtime_->GetInput(index);
+    DLTensor input_tensor = *(arr.operator->());
+    CHECK(tensor->ctx.device_type == input_tensor.ctx.device_type)
```

Comment:
Should we change it to a warning? I would like to know if im doing something inefficient thats causing cross-device copies.

Reply:
I see, I don't think it would be good to have a warning everytime it is called. Maybe we can add a new api to get the device type for an input (or just get the DLTensor)? That way the application can do the check.

Reply:
we do this already here:
https://github.com/neo-ai/neo-ai-dlr/blob/139ab3089f9eb9639522c701a3158755e2212c88/src/dlr_tvm.cc#L156-L161